File: src/types/product.ts
export interface PriceInfo {
  amount: number
  originalAmount?: number
  currency: string
  discountPercentage?: number
}

export interface BulkPrice {
  minQuantity: number
  maxQuantity?: number
  pricePerUnit: number
}

export interface ProductVariant {
  id: string
  name: string
  type: 'color' | 'size' | 'storage' | 'material' | 'other'
  value: string
  colorCode?: string
  priceModifier?: number
  stock: number
  sku: string
}

export interface ProductImage {
  id: string
  url: string
  alt: string
  isPrimary?: boolean
}

export interface ProductSpecification {
  label: string
  value: string
  group?: string
}

export interface Product {
  id: string
  name: string
  slug: string
  description: string
  shortDescription?: string
  price: PriceInfo
  bulkPrices?: BulkPrice[]
  images: ProductImage[]
  category: Category
  subCategory?: string
  brand?: string
  variants?: ProductVariant[]
  specifications?: ProductSpecification[]
  stock: number
  sku: string
  rating: number
  reviewCount: number
  isFeatured?: boolean
  isNew?: boolean
  isOnSale?: boolean
  isBestSeller?: boolean
  tags?: string[]
  warranty?: string
  deliveryInfo?: string
  createdAt: string
  updatedAt: string
}

export interface Category {
  id: string
  name: string
  slug: string
  description?: string
  image?: string
  icon?: string
  parentId?: string
  productCount?: number
  children?: Category[]
}

export interface FilterOption {
  id: string
  label: string
  value: string
  count?: number
}

export interface FilterGroup {
  id: string
  label: string
  type: 'checkbox' | 'radio' | 'range' | 'color'
  options: FilterOption[]
}

export interface PriceRange {
  min: number
  max: number
}

export interface ProductFilters {
  categories?: string[]
  brands?: string[]
  priceRange?: PriceRange
  colors?: string[]
  sizes?: string[]
  ratings?: number[]
  inStock?: boolean
  onSale?: boolean
}

export type SortOption =
  | 'relevance'
  | 'price-low-high'
  | 'price-high-low'
  | 'newest'
  | 'rating'
  | 'best-selling'

export interface ProductListParams {
  page: number
  limit: number
  sort: SortOption
  filters: ProductFilters
  search?: string
}

export interface ProductListResponse {
  products: Product[]
  total: number
  page: number
  totalPages: number
  filters: FilterGroup[]
}

File: backend/VgoApi/Models/Product.cs
namespace VgoApi.Models;

public class Product
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public required string Slug { get; set; }
    public required string Description { get; set; }
    public string? ShortDescription { get; set; }
    public Guid CategoryId { get; set; }
    public string? SubCategory { get; set; }
    public string? Brand { get; set; }
    public int Stock { get; set; }
    public required string Sku { get; set; }
    public decimal Rating { get; set; }
    public int ReviewCount { get; set; }
    public bool IsFeatured { get; set; }
    public bool IsNew { get; set; }
    public bool IsOnSale { get; set; }
    public bool IsBestSeller { get; set; }
    public string? Warranty { get; set; }
    public string? DeliveryInfo { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public Category Category { get; set; } = null!;
    public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
    public ICollection<ProductVariant> Variants { get; set; } = new List<ProductVariant>();
    public ICollection<ProductSpecification> Specifications { get; set; } = new List<ProductSpecification>();
    public ICollection<ProductPrice> Prices { get; set; } = new List<ProductPrice>();
    public ICollection<BulkPrice> BulkPrices { get; set; } = new List<BulkPrice>();
    public ICollection<Review> Reviews { get; set; } = new List<Review>();
    public ICollection<WishlistItem> WishlistItems { get; set; } = new List<WishlistItem>();
}

public class ProductPrice
{
    public Guid Id { get; set; }
    public Guid ProductId { get; set; }
    public decimal Amount { get; set; }
    public decimal? OriginalAmount { get; set; }
    public required string Currency { get; set; }
    public int? DiscountPercentage { get; set; }
    public bool IsActive { get; set; }
    public DateTime ValidFrom { get; set; }
    public DateTime? ValidUntil { get; set; }

    // Navigation properties
    public Product Product { get; set; } = null!;
}

public class ProductImage
{
    public Guid Id { get; set; }
    public Guid ProductId { get; set; }
    public required string Url { get; set; }
    public required string AltText { get; set; }
    public bool IsPrimary { get; set; }
    public int SortOrder { get; set; }
    public DateTime CreatedAt { get; set; }

    // Navigation properties
    public Product Product { get; set; } = null!;
}

public class ProductVariant
{
    public Guid Id { get; set; }
    public Guid ProductId { get; set; }
    public required string Name { get; set; }
    public required string Type { get; set; } // color, size, storage, material, other
    public required string Value { get; set; }
    public string? ColorCode { get; set; }
    public decimal PriceModifier { get; set; }
    public int Stock { get; set; }
    public required string Sku { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public Product Product { get; set; } = null!;
}

public class ProductSpecification
{
    public Guid Id { get; set; }
    public Guid ProductId { get; set; }
    public required string Label { get; set; }
    public required string Value { get; set; }
    public string? GroupName { get; set; }
    public int SortOrder { get; set; }

    // Navigation properties
    public Product Product { get; set; } = null!;
}

public class BulkPrice
{
    public Guid Id { get; set; }
    public Guid ProductId { get; set; }
    public int MinQuantity { get; set; }
    public int? MaxQuantity { get; set; }
    public decimal PricePerUnit { get; set; }

    // Navigation properties
    public Product Product { get; set; } = null!;
}
